### PR TITLE
[ci] Fix windows + mac requirements

### PR DESCRIPTION
**Pull request by krfricke**
Signed-off-by: Kai Fricke <kai@anyscale.com>

## Why are these changes needed?

After #29375 it seems that Windows builds fail for matplotlib installs, and mac builds don't find gradio + latest boto releases. This PR adjusts the requirements files to fix this.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/requirements.txt [hunk 1]
```diff
@@ -40,11 +40,11 @@ kopf; python_version >= '3.7'
 scipy
 aiohttp>=3.7
 fastapi
-gym<0.24.0,>=0.21.0
+gym<0.24.1,>=0.21.0
 opencensus
 fsspec
 dm_tree
-matplotlib!=3.4.3
+matplotlib!=3.4.3; platform_system != "Windows"
```
Comment (amogkam) [inline python/requirements.txt:47]: Do we need to make the corresponding changes in setup.py?

**Review by amogkam — APPROVED** [review]

File: python/requirements.txt (hunk 1 above)
Comment (krfricke) [reply python/requirements.txt:47]: No, this should be very CI specific - the CI machines don't have windows fonts installed and thus the environment variable lookup fails.

Comment (krfricke) [reply python/requirements.txt:47]: (a regular windows install will have fonts installed)

Comment (amogkam) [reply python/requirements.txt:47]: Got it! How about for the gym change?

Comment (krfricke) [reply python/requirements.txt:47]: So to be clear, for windows we don't want to install matplotlib at all

Comment (amogkam) [reply python/requirements.txt:47]: Yep sounds good! What about the change to gym?

**Comment by krfricke** [comment]
Windows + Doc tests are passing. Mac is backlogged, so I'll merge for now (current queue is 24 hours+).

**Pull request merged**